teps/0093-tkn-cli-extension-for-signverify.md

```diff
@@ -0,0 +1,347 @@
+
+Objective in this proposal is to faciliate an easy and handy approach for task/pipeline providers to sign their artifacts and for consumers to verify them using native `tkn cli` commands.
+
+We have an open-source reference implementation available at: [tap8stry-pipelines](https://github.com/tap8stry/tapestry-pipelines).
+
+Also, this complements another great [TEP-0091](https://github.com/tektoncd/community/blob/43643c958fe9cdc76b847996a37d83836656e8eb/teps/0091-verified-task-bundles.md). In TEP-0091, the proposal is to allow sign verification and enforcement at the controller. While in this proposal, we are suggesting an CLI extension for developers to sign and verify pipelines locally at composition. Also, allow sign/verification for plain YAMLs as well as bundles.
+
+## Motivation
+
+There is an on-going efforts for ensuring software supply chain harderning for our applications, where we discover all dependencies and verify their signatures for integrity assurance. Just like our application, our tekton CI/CD pipelines are also being composed of open-source third-party dependencies. For instance, using off the shelf task or pipeline definition from open catalogs. As we bring-in and embed such dependencies in our pipelines, we need to ensure their integrity and identity. 
+
+![](https://i.imgur.com/GRLlm4L.png)
+
+For instance, consider sample pipeline shown above. It is composed by leveraging open-source task definitions for `git-clone`, `vulnerability-scan` and `build-image`. In this pipeline, assume that the `git-clone` task is compromised in the sense that it tampers with the original source artifacts during `clone` operation. Since, our source artifacts are not tampered, and remainer of the tasks in the pipeline operates on these  artifacts, our whole pipeline is now compromised. 
+
+Also, we usually hand-over various credentials to various tasks in the pipeline. For instance, our github OAUTH token to `git-clone` task, or our registry credentials to `build-and-publish` task. Some compromised tasks could even steal these credentials. 
+
+Therefore, we should faciliate open, secure and integral process for developers to share and distribute task/pipeline definitions. 
+
+### Goals
+
+1. Build `tkn` native experience for developers ( providers and users) to sign/verify shared resources like task or pipeline definitions through CLI
+2. Perform signing at different granularities that includes pipeline specs, task specs, step images, bundles etc. 
+3. Optional feature to convert tkn definitions to `intoto` layouts and provide layout signing
```

Comment:
sounds like this could be part of the [TEP-0091 about in-toto layout creation and verification in the CLI](https://github.com/tektoncd/community/pull/534)  ?